src/linter/ui5Types/amdTranspiler/tsTransformer.ts

```diff
@@ -112,6 +144,55 @@ function transform(
 						throw err;
 					}
 				}
+			} else {
+				try {
+					let variableStatement: ts.VariableStatement | undefined;
+					let className: string | undefined;
+
+					// Check if class is assigned to a variable.
+					// If so, use the local variable name as class name and remove the variable declaration
+					if (
+						ts.isVariableDeclaration(node.parent) &&
+						ts.isVariableDeclarationList(node.parent.parent) &&
+						ts.isVariableStatement(node.parent.parent.parent)
+					) {
+						variableStatement = node.parent.parent.parent;
+						className = node.parent.name.getText();
+					}
+
+					// For now, only rewrite extend calls in expressions and variable statements
+					if (variableStatement || ts.isExpressionStatement(node.parent)) {
+						const classDeclaration = rewriteExtendCall(nodeFactory, node, undefined, className);
+						if (variableStatement) {
+							// We can't replace the variable declaration with the class declaration (not valid),
```

Comment:
Should be moved into the next if, right? I.e. it doesn't apply to the else case

Reply:
True, thanks. I've update the comments.